pkg/arvo/app/chat-hook.hoon

```diff
@@ -157,14 +181,61 @@
 ++  peer-mailbox
   |=  pax=path
   ^-  (quip move _this)
+  ?>  ?=(^ pax)
+  =/  last  (dec (lent pax))
+  =/  backlog-start=(unit @ud)
+    %+  rush
+      (snag last `(list @ta)`pax)
+    dem:ag
+  =>  .(pax `path`(oust [last last] `(list @ta)`pax))
   ?>  ?=([* ^] pax)
   ?>  (~(has by synced) pax)
   ::  scry permissions to check if read is permitted
   ?>  (permitted-scry [(scot %p src.bol) %chat (weld pax /read)])
   =/  box  (chat-scry pax)
   ?~  box  !!
   :_  this
-  [ost.bol %diff %chat-update [%create (slav %p i.pax) pax]]~
+  :-  [ost.bol %diff %chat-update [%create (slav %p i.pax) pax]]
+  ?:
+    ?&
+      ?=(^ backlog-start)
+      (~(got by allow-history) pax)
+    ==
+    (paginate-messages pax u.box u.backlog-start)
```

Comment:
Why do we paginate instead of just sending whatever's requested? I don't think we fall over on big messages anymore.

Reply:
Idk... Big messages scare me.

Reply:
Pagination is generally righteous IMO, at least in cases where a client can practically make use of the individual pages. But there also aren't any problems with big messages that I'm aware of.

Reply:
_Maybe_ this is unresolved since pagination here is a data-sending mechanism rather than part of the data-retrieving query. Since big messages aren't a problem, maybe just pull this logic out and send backlog all at once, to reduce complexity. But I don't feel super strongly.